proxy/mempool/mempool_schedule.py

```diff
@@ -390,6 +432,8 @@ def add_tx(self, tx: MPTxRequest) -> MPTxSendResult:
                 )
                 self._drop_tx_from_sender_pool(sender_pool, old_tx)
 
+        tx.update_heartbeat()
```

Comment:
The heartbeat is already set in the ctor of MPTxRequest

Reply:
For the "new" design, we should always update the heartbeat on either:

  - insertion
  - fee replacement

This update here catches both scenarios. If we do not do this here, then the only time the heartbeat is set is during construction.